simtools/applications/generate_corsika_histograms.py

```diff
@@ -0,0 +1,425 @@
+        if png:
+            figure, figure_name = corsika_output_visualize.plot_1D_event_header_distribution(
+                instance, event_header_element
+            )
+            output_file_name = Path(output_dir).joinpath(figure_name)
+            logger.info(f"Saving histogram to {output_file_name}")
+            figure.savefig(output_file_name, bbox_inches="tight")
+        if ecsv:
+            instance.export_event_header_1D_histogram(
+                event_header_element, bins=50, hist_range=None
+            )
+
+
+def _run_event_2D_histograms(instance, output_dir, event_2D_header_keys, png, ecsv):
+    """
+    Auxiliary function to run the histograms for the arguments given by event_1D_histograms.
+
+    Parameters
+    ----------
+    instance: `CorsikaOutput` instance.
+        The CorsikaOutput instance created in main.
+    output_dir: str
+        The output directory where to save the histograms.
+    event_2D_header_keys: str
+        Produce 1D histograms for elements given in `event_1D_header_keys` from the CORSIKA event
+        header and save into ecsv/png files.
+    png: bool
+        If true, histograms are saved into png files.
+    ecsv: bool
+        If true, histograms are saved into ecsv files.
+    """
+    for i_event_header_element, _ in enumerate(event_2D_header_keys[::2]):
+        if png:
+            figure, figure_name = corsika_output_visualize.plot_2D_event_header_distribution(
+                instance,
+                event_2D_header_keys[i_event_header_element],
+                event_2D_header_keys[i_event_header_element + 1],
+            )
+            output_file_name = Path(output_dir).joinpath(figure_name)
+            logger.info(f"Saving histogram to {output_file_name}")
+            figure.savefig(output_file_name, bbox_inches="tight")
+        if ecsv:
+            instance.export_event_header_2D_histogram(
+                event_2D_header_keys[i_event_header_element],
+                event_2D_header_keys[i_event_header_element + 1],
+                bins=50,
+                hist_range=None,
+            )
+
+
+def main():
+
+    label = Path(__file__).stem
+    description = "Generate histograms for the Cherenkov photons saved in the CORSIKA IACT file."
+    usage = ""
+    args_dict, _ = _parse(label, description, usage)
+
+    io_handler_instance = io_handler.IOHandler()
+
+    logger.setLevel(gen.get_log_level_from_user(args_dict["log_level"]))
+    initial_time = time.time()
+    logger.info("Starting the application.")
+
+    instance = CorsikaOutput(args_dict["IACT_file"])
+    if args_dict["telescope_indices"] is not None:
+        indices = np.array(args_dict["telescope_indices"]).astype(int)
+    else:
+        indices = None
+    instance.set_histograms(
+        telescope_indices=indices,
+        individual_telescopes=args_dict["individual_telescopes"],
+        hist_config=args_dict["hist_config"],
+    )
+
+    if args_dict["output_directory"] is None:
```

Comment:
I adjusted it to use the output_path from the configuration module directly in the application. I do not think it is necessary to add more arguments to the module since it is not strictly necessary. Unless we end up with >3 applications using the corsika_output module and all of them need the output path, then this change would be justified. Also, I do not know which modules you refer to (there are a handful of corsika and simtelarray files)

Reply:
OK this is anyway removed now so probably irrelevant.